src/main/scala/com/sumologic/sumobot/core/Receptionist.scala

```diff
@@ -110,43 +110,49 @@ class Receptionist(rtmClient: SlackRtmClient,
       pendingIMSessionsByUserId = pendingIMSessionsByUserId + (userId -> (doneRecipient, doneMessage))
 
     case message: Message if !tooOld(message.ts, message) =>
-      translateAndDispatch(message.channel, message.user, message.text)
+      translateAndDispatch(message.channel, message.user, message.text, message.ts)
```

Comment:
Wondering: Would it make sense to pass `message` into `translateAndDispatch()` and read the four fields there?

Reply:
`translateAndDispatch` is called for both `Message` and `MessageChanged` events which although are similar, they don't share a superclass.